### DISCUSSION TITLE
How to activate application from android foreground service notification

### DISCUSSION THREAD

**Discussion by vkotov** [AvaloniaUI/Avalonia#12602] [Q&A]
Hi

I create a foreground service in Avalona android application from examples for MAUI and it works perfectly.
But I need to activate my application by tapping on notification bar. So my code for execute foreground service is following:

```
public override StartCommandResult OnStartCommand(Intent? intent, StartCommandFlags flags, int startId)
{
    var mainActivityIntent = new Intent(this, typeof(MainActivity));
    var pendingIntent = PendingIntent.GetActivity(this, 0, mainActivityIntent, PendingIntentFlags.UpdateCurrent | PendingIntentFlags.Immutable);
    
    var notification = new NotificationCompat.Builder(this, MainActivity.ChannelId)
	    .SetContentTitle("My Service")
	    .SetContentText("Foreground Service is running")
	    .SetSmallIcon(Resource.Drawable.Icon)
	    .SetContentIntent(pendingIntent)  //TODO: Unable to activate application when on foreground
	    .SetAutoCancel(false)
	    .SetOngoing(true)
	    .Build();
    
    // Don't forget for Foreground service need to update AndroidManifest.xml with <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
    StartForeground(1, notification);
    
    return StartCommandResult.NotSticky;
}
```

This code work when application executed but not on screen. So when I tap on notification bar then application going to screen successfully. But If application already active (on screen) I have exception:

```
System.NotSupportedException
  Message=Unable to activate instance of type Avalonia.Android.AvaloniaView from native handle 0x7fded656d4 (key_handle 0xa655207).

```

Avalonia 11.0.4 is used

What I doing wrong? 
Do you have an example with android foreground service in Avalonia?

**Comment by timunie** [comment]
@vkotov can you link or attach a minimal sample to reproduce? then maybe someone has an idea or we find it's a bug or missing feature. 

Thx.

**Reply by vkotov** [reply]
@timunie can you try execute my example under android device or emulator? 
[AvaloniaAndroidForegroundService.zip](https://github.com/AvaloniaUI/Avalonia/files/12399196/AvaloniaAndroidForegroundService.zip)

Tap notification bar when application on screen.
[tap.webm](https://github.com/AvaloniaUI/Avalonia/assets/1422494/752a9fba-3d64-402a-be4a-f60a9de50496)

![exception](https://github.com/AvaloniaUI/Avalonia/assets/1422494/b8a7c04b-ab63-4d57-a242-263b0aa09070)

Thanks

**Reply by timunie** [reply]
@emmauss do you have any idea running the attached sample?
